### ISSUE TITLE
Problem When Exporting A Project Using E羊icques

### ISSUE THREAD

**Issue by Rhet0rical** [SheepTester/scratch-gui#59]
When exporting a project, I usually open another client to see if it works and this time I came up with this error.
(its the same as the last one)

The same problem still persists, so until there is a patch of some sorts I would like to know how to fix the problem.
File : (https://github.com/SheepTester/htmlifier/files/6951722/Predat0r_1.2.11_RAW.zip)

**Comment by Rhet0rical** [comment]
Apologies. I was unable to find the correct repository so I supposed the other one was the correct one.

**Comment by SheepTester** [comment]
(ref: SheepTester/htmlifier#105)

Does it happen with all sounds you upload? I was not able to reproduce the bug by adding a random wav file to your project

**Comment by Rhet0rical** [comment]
It does not, it seems like it happens at random.

**Comment by Rhet0rical** [comment]
How did you fix it last time?

**Comment by Rhet0rical** [comment]
Although I did happen to delete the sound and replace it with a new one, the problem still persists.

**Comment by Rhet0rical** [comment]
[Predat0r_1.2.10_RAW.zip](https://github.com/SheepTester/scratch-gui/files/6951756/Predat0r_1.2.10_RAW.zip)
There is the replacement for the file I posted earlier.

**Comment by Rhet0rical** [comment]
@SheepTester Do you think it is a problem with the sound format I'm using? I'm surprised to find out that every mp3 I import into the engine comes out as a .wav file. Maybe its a problem with the translation.

**Comment by Rhet0rical** [comment]
PS. I do not import .wav files. I import them using mp3s.

**Comment by Rhet0rical** [comment]
I have now downloaded and converted every sound file into a .wav file. Problem Persists :\

**Comment by Rhet0rical** [comment]
@SheepTester

**Comment by SheepTester** [comment]
> [Predat0r_1.2.10_RAW.zip](https://github.com/SheepTester/scratch-gui/files/6951756/Predat0r_1.2.10_RAW.zip)
> There is the replacement for the file I posted earlier.

Were you trying to add the sounds to that project, or is that the same broken project from the earlier issue or a new broken project? The project fails to load for me in the editor.

Here's my procedure:

1. I opened the fixed file, which I linked in https://github.com/SheepTester/htmlifier/issues/105#issuecomment-894874568 (here's a [direct link to the .zip](https://github.com/SheepTester/htmlifier/files/6951626/x.replaced.with.wav.zip), which you can rename to an .sb3), in [E羊icques](https://sheeptester.github.io/scratch-gui/).
2. I uploaded two sounds ([test.mp3](https://cdn.discordapp.com/attachments/775797057926856754/874418948735635487/test.mp3) and [test.wav](https://cdn.discordapp.com/attachments/775797057926856754/874418961335332874/test.wav)) to the first sprite, `Global - Loading`.
3. I saved the project to my computer as an .sb3 file. Here is the result, if you want to verify it yourself: [project.zip](https://github.com/SheepTester/scratch-gui/files/6957956/overwrite.me.zip).

However, I cannot reproduce your bug. I am able to open that project in E羊icques as well as upload it to the [HTMLifier](https://sheeptester.github.io/htmlifier/) with the default settings, and the resulting HTML file loads and runs the project properly.

**Comment by Rhet0rical** [comment]
Weird.
1. Do I Have To Put Both The MP3 And The WAV File Into The Sprite For It To Work?
2. It Has To Be A Problem With The Translation, I Can't See Otherwise. Although It Is Weird That You Can't Reproduce The Bug, I Replaced Every Sound File (taking the mp3 and translating it into a wav in audacity) It Produced A .X File For Almost Every .Wav File That Was Inside The Folder. It Happens At Random.

**Comment by Rhet0rical** [comment]
For Some Reason, The Old Version When Unzipping It It Shows That Every Sound File Is A .Wav When I Imported The Files Orginally As An Mp3

**Comment by SheepTester** [comment]
> For Some Reason, The Old Version When Unzipping It It Shows That Every Sound File Is A .Wav When I Imported The Files Orginally As An Mp3

Scratch might convert the mp3 files to wav

> Weird.
> 
> 1. Do I Have To Put Both The MP3 And The WAV File Into The Sprite For It To Work?
> 2. It Has To Be A Problem With The Translation, I Can't See Otherwise. Although It Is Weird That You Can't Reproduce The Bug, I Replaced Every Sound File (taking the mp3 and translating it into a wav in audacity) It Produced A .X File For Almost Every .Wav File That Was Inside The Folder. It Happens At Random.

1. I put both an mp3 and wav to see if either file type would cause the bug. I repeated the procedure, but I only uploaded the mp3 file. The modified project still loaded fine.
2. Which translation are you referring to? Converting the mp3 to wav? Or when saving the project to your computer? Or HTMLifying it?

What OS/browser are you using? Are you using any browser extensions?

**Comment by Rhet0rical** [comment]
If the project loaded fine, maybe scratch requires creates a copy of the file and translates that to a wav.

And I'm referring to the translation with the sound files.

**Comment by Rhet0rical** [comment]
I Use Chromium And I Run Without Any Extensions

**Comment by Rhet0rical** [comment]
Im Pretty Sure You Know This, But Changing The File Extension For The .Wav (If It Matches The Accompanied .X File) And Changing That To A .Wav File While Deleting The Other One, It Seems Like Its The Same As The Original File, Just Corrupted Maybe.

**Comment by SheepTester** [comment]
Yes, the .x file contains the same data as the .wav file. This is bizarre.

I see that your project loads normally in vanilla Scratch. Can you try uploading sounds using the vanilla editor (either on the Scratch website or with the [beta](https://llk.github.io/scratch-gui/develop/)) to see if the problem is with this mod or with Scratch in general?

**Comment by Rhet0rical** [comment]
Sure. Would You Like Me To Send The File?

**Comment by SheepTester** [comment]
Since I'm not able to reproduce the bug, I think it is best if you try it yourself, either on the Scratch website's editor or on https://llk.github.io/scratch-gui/develop/

**Comment by Rhet0rical** [comment]
The Problem Didn't Happen On Vanilla Scratch. Although It Does Have A Problem Saving.

I Keep On Getting An Error When I Upload The File, I Have It Uploaded On Drive.
https://drive.google.com/file/d/1_KAGWuQMx9i2K11llBaY3hL7LZlrYN77/view?usp=sharing

**Comment by SheepTester** [comment]
> The Problem Didn't Happen On Vanilla Scratch. Although It Does Have A Problem Saving.
> 
> I Keep On Getting An Error When I Upload The File, I Have It Uploaded On Drive.
> https://drive.google.com/file/d/1_KAGWuQMx9i2K11llBaY3hL7LZlrYN77/view?usp=sharing

Well, if you aren't able to save the project, then I don't think you can conclude that the problem doesn't happen in vanilla Scratch.

You might have to be patient when saving the project to your computer since it's quite large. It took at least 30 seconds before the download started, and I'm using a somewhat powerful computer. You can check the console to see if there are any errors; if there's nothing, then it's probably still compressing the project into an .sb3 file

**Comment by Rhet0rical** [comment]
Have you found out the problem yet? Is it something that I'm doing? Or does the engine just have a deep hate for .mp3s

**Comment by Rhet0rical** [comment]
You Mean On Scratch? Sure. I'll Send A Screenshot If I Get Anything

**Comment by Rhet0rical** [comment]
On scratch i get no console errors, Only The Basic "STOP" message.

**Comment by Rhet0rical** [comment]
I can try to upload mp3s again to replace all of the other files to see if that does anything.

**Comment by Rhet0rical** [comment]
It Worked!

**Comment by Rhet0rical** [comment]
It Seems Like The Engine Has A Problem With .Wav Files. Replacing All The Sounds With Mp3s Seemed To Work.

**Comment by Rhet0rical** [comment]
Although It Does Seem Like If There Is A Sound File Over 2K Kilobytes, It Translates That To A .Wav File.

**Comment by SheepTester** [comment]
I think it uses your browser to convert mp3 to wav

**Comment by Rhet0rical** [comment]
Howcome?

**Comment by Rhet0rical** [comment]
There's An Offline Build For E羊icques But I Don't Know How To Launch It. Im Also Worried I Can't Enable Tools And Change The Settings In The Offline Build.

**Comment by Rhet0rical** [comment]
Where do you think the .x File comes from? Is it some type of residue or failed translation of the original .mp3?

**Comment by SheepTester** [comment]
> There's An Offline Build For E羊icques But I Don't Know How To Launch It. Im Also Worried I Can't Enable Tools And Change The Settings In The Offline Build.

Open index.html in the editor

> Where do you think the .x File comes from? Is it some type of residue or failed translation of the original .mp3?

No idea, it hasn't happened before it seems

> Howcome?

Dealing with mp3's apparently involves some patented technology, so for legal reasons, it's better to rely on the browser than include patented code in Scratch. Also because it's better to use the browser if the browser can do it to not have to send code that does what the browser can already do over the internet, which would increase loading times

**Comment by Rhet0rical** [comment]
The export receives no errors or any of those pesky .X files when using only .mp3 sound files, so something is fixed. Still doesn't explain the .x file though.
